lib/src/ast/css/README.md

```diff
@@ -0,0 +1,50 @@
+# CSS Abstract Syntax Tree
+
+This directory contains the abstract syntax tree that represents a plain CSS
+file generated by Sass compilation. It differs from other Sass ASTs in two major
+ways:
+
+1. Instead of being created by [a parser], it's created by [the evaluator] as it
+   traverses the [Sass AST].
+
+   [a parser]: ../../parse/README.md
+   [the evaluator]: ../../visitor/async_evaluate.dart
+   [Sass AST]: ../sass/README.md
+
+2. Because of various Sass features like `@extend` and at-rule hoisting, the CSS
+   AST is mutable even though all other ASTs are immutable.
+
+**Note:** the CSS AST doesn't have its own representation of declaration values.
+Instead, declaration values are represented as [`Value`] objects. This does mean
+that a CSS AST can be in a state where some of its values aren't representable
+in plain CSS (such as maps)—in this case, [the serializer] will emit an error.
+
+[`Value`]: ../../value/README.md
+[the serializer]: ../../visitor/serialize.dart
+
+## Mutable and Immutable Views
+
+Internally, the CSS AST is mutable to allow for operations like hoisting rules
+to the root of the AST and updating existing selectors when `@extend` rules are
+encountered. However, because mutability poses a high risk for "spooky action at
+a distance", we limit access to mutating APIs exclusively to the evaluator.
```

Comment:
Would it make sense to link to https://en.wikipedia.org/wiki/Action_at_a_distance_(computer_programming) or something similar here for anyone who's unfamiliar with the phrase?